### DISCUSSION TITLE
Confusing change in behavior for 'less', a question and some thoughts...

### DISCUSSION THREAD

**Discussion by wtanksleyjr** [wezterm/wezterm#5585] [Q&A]
### What Operating System(s) are you running on?

Windows

### Which Wayland compositor or X11 Window manager(s) are you using?

I'm muxing from Windows to Linux; on the Linux I'm running wezterm 20240203-110809-5046fc22 (but I'm running headless there so no X11 or Wayland).

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

When I started using wezterm to mux into Linux (BTW completely awesome feature), `less` unexpectedly changed its behavior. The new behavior is to act as though I'd run `less --quit-if-one-screen`, and it also changed its format so there are line numbers and surprisingly sophisticated formatting. I honestly really like the new formating, but I simply cannot STAND the --quit-if-one-screen behavior. The problem is that less doesn't seem to have a way to turn that OFF, and I cannot find any place where it's being turned ON so I can disable it.

After hours of research brought me from "I have no idea this feature exists" to "OK less can do this when people ask it to, but I'm not asking", I'm thinking this MIGHT have something to do with wezterm's "alternate screen" feature. Maybe. Does it?

Does anyone know? Has anyone seen this before?

**Comment by wez** [comment]
It's probably not wezterm, it's probably something in your shell environment. Keep in mind that wezterm always spawns login shells.

**Comment by wtanksleyjr** [comment]
Thanks, that IS always a possibility - the fact that it started when I
started using wezterm doesn't mean wezterm caused it.

Tempted to give up; finding that there's a third dimension to this I have
to completely understand before I can even start knowing what to look for
... Two possible interactions might be my limit, adding a third possible
one means I can't possibly tell where to begin reading. I remember the
difference between login shells and whatever else they're called nearly
broke me when I used the Mac.

Ah well, it's not much of a decision, the new undesirable behavior is not
as bad as the rest of wezterm is good.

-Wm
